src/cross_validate.py

```diff
@@ -12,13 +13,17 @@
 
 
 class CrossValidation:
-    models: List[BaseSolution] = []
     device = 'cuda' if torch.cuda.is_available() else 'cpu'
 
-    def __init__(self, n_splits: int = 5):
+    def __init__(self, saving_dir: Path, n_splits: int = 5):
```

Comment:
I think it's wiser to just allow the user give a str here, instead of a Path variable. You can do it like this:
```
def __init__(self, saving_dir: str, n_splits: int = 5):
    self.saving_dir = Path(saving_dir)
    self.saving_dir.mkdir(exist_ok=True)
```

Reply:
I'm follower of path lib library, but ok, I will change it